routing/index_router.cpp

```diff
@@ -558,29 +592,39 @@ bool IndexRouter::FindBestSegment(m2::PointD const & point, m2::PointD const & d
   vector<pair<Edge, Junction>> candidates;
   m_roadGraph.FindClosestEdges(point, kMaxRoadCandidates, candidates);
 
-  auto const getSegmentByEdge = [&numMwmId](Edge const & edge){
-      return Segment(numMwmId, edge.GetFeatureId().m_index, edge.GetSegId(), edge.IsForward());
+  auto const getSegmentByEdge = [&numMwmId](Edge const & edge, bool isForward) {
+    return Segment(numMwmId, edge.GetFeatureId().m_index, edge.GetSegId(), isForward);
   };
 
   // Getting rid of knowingly bad candidates.
   my::EraseIf(candidates, [&](pair<Edge, Junction> const & p){
     Edge const & edge = p.first;
-    return edge.GetFeatureId().m_mwmId != mwmId || IsDeadEnd(getSegmentByEdge(edge), isOutgoing, worldGraph);
+    return edge.GetFeatureId().m_mwmId != mwmId ||
+           IsDeadEnd(getSegmentByEdge(edge, edge.IsForward()), isOutgoing, worldGraph);
   });
 
   if (candidates.empty())
     return false;
 
-  BestEdgeComparator bestEdgeComparator(point, direction);
-  Edge & bestEdge = candidates[0].first;
+  BestEdgeComparator bestEdgeComparator(point, direction, [this](FeatureID const & featureId) {
+    return this->IsOneWayFeature(featureId);
+  });
+  Edge bestEdge = candidates[0].first;
```

Comment:
Да, ты прав. Если два сегмента, о которых ты пишешь, будут двунаправленные, то 
```c++
bool IndexRouter::BestEdgeComparator::IsAlmostConformed(Edge const & edge) const
{
  bool forward = true;
  if (!IsAlmostCollinear(edge, forward))
    return false;

  // Note. In case of oneway edges |edge| is conformed to |m_direction| iff vector of |edge|
  // and vector |m_direction| are almost collinear and have the same direction.
  return m_isOnewayFn(edge.GetFeatureId()) ? forward : true;
}
```
всегда вернет true. 

Я поправлю эту ошибку при планируемых выше изменениях. Спасибо!